src/pyclts/commands/map.py

```diff
@@ -1,96 +1,121 @@
-                match = list(bipa._regex.finditer(rg))
-                if len(match) == 2:
-                    s1 = bipa[rg[:match[1].start()]]
-                    s2 = bipa[rg[match[1].start():]]
-                    if s1.type == 'consonant' and s2.type == \
-                            'consonant':
-                        # check for prenasalized stuff
-                        if s1.manner == 'nasal' and (
-                                s2.place == s2.place or \
-                                        s2.manner in ['stop', 'affricate',
-                                            'fricative', 'implosive']):
-                            data[i][bidx] = '*ⁿ'+s2.s
+                sound = bipa[raw_grapheme]
+                if sound.type == "unknownsound":
+                    match = list(bipa._regex.finditer(raw_grapheme))
+                    if len(match) == 2:
+                        sound1 = bipa[raw_grapheme[: match[1].start()]]
+                        sound2 = bipa[raw_grapheme[match[1].start() :]]
+                        if sound1.type == "consonant" and sound2.type == "consonant":
+                            # check for prenasalized stuff
+                            if sound1.manner == "nasal" and (
+                                sound2.place == sound2.place
+                                or sound2.manner
+                                in ["stop", "affricate", "fricative", "implosive"]
+                            ):
+                                row["BIPA"] = "*ⁿ" + str(sound2)
+                            else:
+                                row["BIPA"] = "(?)"
                         else:
-                            data[i][bidx] = '?'
-                            count += 1
+                            row["BIPA"] = "<NA>"
                     else:
-                        data[i][bidx] = '<NA>'
-                        count += 1
-                else:
-                    data[i][bidx] = '<NA>'
-                    count += 1
-            elif s.type == 'marker':
-                data[i][bidx] = rg
-            elif s.type == 'cluster':
-                # check for prenasalized stuff
-                if s.from_sound.manner == 'nasal' and (
-                        s.from_sound.place == s.to_sound.place or \
-                                s.to_sound.manner in ['stop', 'affricate',
-                                    'fricative', 'implosive']):
-                    data[i][bidx] = '*ⁿ'+s.to_sound.s
-                elif s.to_sound.manner == 'fricative' and s.from_sound.manner == 'stop':
-                    ns = bipa[s.to_sound.name.replace('fricative',
-                        'affricate')]
-                    if ns.type == 'consonant':
-                        data[i][bidx] = '*'+ns.to_sound.s
+                        row["BIPA"] = "<NA>"
+                elif sound.type == "marker":
+                    row["BIPA"] = raw_grapheme
+                elif sound.type == "cluster":
```

Comment:
Note that we may add more here in the future, by e.g., converting a `tr` to `t+superscript r`, but it is fine for now.